balancer-js/src/modules/pools/pool-types/concerns/metaStable/priceImpact.concern.ts

```diff
@@ -16,51 +16,20 @@ export class MetaStablePoolPriceImpact implements PriceImpactConcern {
   /**
    * Calculates the BPT return amount when investing with no price impact.
    * @param { Pool } pool Investment pool.
-   * @param { string [] } amounts Token amounts being invested. Needs a value for each pool token.
+   * @param { string [] } tokenAmounts Token amounts being invested. Needs a value for each pool token.
    * @returns { string } BPT amount.
    */
   bptZeroPriceImpact(pool: Pool, tokenAmounts: bigint[]): bigint {
     if (tokenAmounts.length !== pool.tokensList.length)
       throw new BalancerError(BalancerErrorCode.INPUT_LENGTH_MISMATCH);
 
-    const {
-      parsedBalances,
-      parsedDecimals,
-      parsedPriceRates,
-      parsedAmp,
-      parsedTotalShares,
-    } = parsePoolInfo(pool);
-    const totalShares = BigInt(parsedTotalShares);
-    const decimals = parsedDecimals.map((decimals) => {
-      if (!decimals)
-        throw new BalancerError(BalancerErrorCode.MISSING_DECIMALS);
-      return BigInt(decimals);
-    });
-    const priceRates = parsedPriceRates.map((rate) => {
-      if (!rate) throw new BalancerError(BalancerErrorCode.MISSING_PRICE_RATE);
-      return BigInt(rate);
-    });
-    if (!parsedAmp)
-      throw new BalancerError(BalancerErrorCode.MISSING_PRICE_RATE);
+    const { priceRates, ampWithPrecision, totalSharesEvm, upScaledBalances } =
+      parsePoolInfo(pool);
 
-    const scalingFactors = decimals.map((decimals) =>
-      _computeScalingFactor(BigInt(decimals))
-    );
-    const balances = parsedBalances.map((balance, i) =>
-      _upscale(BigInt(balance), scalingFactors[i])
-    );
-    const balancesScaled = balances.map((balance, i) =>
-      SolidityMaths.mulDownFixed(balance, priceRates[i])
-    );
     let bptZeroPriceImpact = BZERO;
-    for (let i = 0; i < balances.length; i++) {
+    for (let i = 0; i < upScaledBalances.length; i++) {
       const price =
-        (bptSpotPrice(
-          BigInt(parsedAmp as string), // this already includes the extra digits from precision
-          balancesScaled,
-          totalShares,
-          i
-        ) *
+        (bptSpotPrice(ampWithPrecision, upScaledBalances, totalSharesEvm, i) *
```

Comment:
There might be an issue using `upScaledBalances` here? (As they will all be 1e18)

Reply:
They were already scaled on the previous version of the code. I simply replaced local logic with what was already implemented within parsePoolInfo.